benchmark/main.cpp

```diff
@@ -755,6 +756,7 @@ MBED_NOINLINE static int benchmark_ctr_drbg()
     if (ret == MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE) {
```

Comment:
MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE is not a recoverable error, for ctr_drbg, so I don't think it requires special treatment for this error

Reply:
I am not sure what you mean. The special treatment is only so that the console printout shows up nicely in the terminal. Perhaps there is something I am missing?...

Reply:
I think that in all the other `MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE` we should print and skip the specific test, as it's a reasonable error to return.
For ctr_drbg, it means that ctr_drbg will fail, as it uses AES256 only, unless configured to 128 bit, as introduced in https://github.com/ARMmbed/mbedtls/pull/1902
So, this error is not a valid error for ctr_drbg, IMHO.
If it's only for the printout, well, the `PRINT_ERROR` will handle this, no?

Reply:
I still do not understand what the change request is here. We are already printing out the test and the string "Feature Unavailable" when AES is not available. This is equivalent to skip because we are not letting the benchmark application failed for that. I do not see the point of printing the actual error string corresponding to MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE and this would be inconsistent with the checks in other parts of this application

Reply:
Because, when we skip a test, it's a valid error, and the return code of the benchmark should be success.
On ctr_drbg, on the other hand, when `MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE ` is returned, it is a real error, and we should fail the test, and developers should be aware that they will need to modify thei ctr_drbg to use 128 bit key size(after https://github.com/ARMmbed/mbedtls/pull/1902 is merged)

Reply:
Ok, I see what you mean, but I do not agree with it. If ctr_drbg does not run because it has no AES that is not a bug or a real failure according to what we have done in the benchmark application. Its just that the primitive is unavailable under the current configuration. Isnt this comparable to saying that AES-256 is not available so its an actual error because the user should change to something like AES-128?

Reply:
No, because for ctr_drbg, the usage of AES256 is embedded within the algorithm and not configurable, and not under the user's discretion.
After it will be configurable to 128, I still think that it should be an error, and not just skipped, because it's an algorithm using AES, and not just a primitive AES, but I won't insist on it, after it will be configurable

Reply:
It's not configurable to use AES-128, yet. After it is configurable, we can make this an error. Is any further action needed here?

Reply:
Correct, this is not configurable. This is why It is a failure error, that should cause the CI to fail, and we should not have special treatment for this specific error, in case of ctr drbg